tests/main/uc20-create-partitions-encrypt/task.yaml

```diff
@@ -58,58 +75,78 @@ prepare: |
     snap download --channel=20/edge pc
     unsquashfs -d gadget-dir pc_*.snap
 
-    echo "Install EFI binaries"
-    bootdir=/run/mnt/ubuntu-boot/EFI/boot
-    mkdir -p "$bootdir"
-    cp /usr/lib/shim/shimx64.efi.signed "$bootdir"/bootx64.efi
-    cp /usr/lib/grub/x86_64-efi-signed/grubx64.efi.signed "$bootdir"/grubx64.efi
-
 execute: |
-    bootdir=/run/mnt/ubuntu-boot/EFI/boot
-    ls -l "$bootdir"
-    sbverify --list "$bootdir"/bootx64.efi
-    sbverify --list "$bootdir"/grubx64.efi
-    LOOP="$(cat loop.txt)"
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        bootdir=/run/mnt/ubuntu-boot/EFI/boot
+        ls -l "$bootdir"
+        sbverify --list "$bootdir"/bootx64.efi
+        sbverify --list "$bootdir"/grubx64.efi
+        LOOP="$(cat loop.txt)"
+        mount "${LOOP}"p2 /run/mnt/ubuntu-seed
+
+        echo "Run the snap-bootstrap tool"
+        go get ../../lib/uc20-create-partitions
+        uc20-create-partitions \
+            --encrypt --key-file /run/mnt/ubuntu-seed/keyfile \
+            --recovery-key-file /run/mnt/ubuntu-data/system-data/var/lib/snapd/fde/recovery-key \
+            --policy-update-data-file /run/mnt/ubuntu-data/system-data/var/lib/snapd/fde/policy-update-data \
+            --tpm-lockout-auth /run/mnt/ubuntu-data/system-data/var/lib/snapd/fde/tpm-lockout-auth \
+            --model <(snap model --assertion) \
+            ./gadget-dir "$LOOP"
+
+        echo "Check that the key file was created"
```

Comment:
shouldn't we simply share the check across the reboots? or at least do some minimal checks about p4/ubuuntu-data in particular also in the reinstall case?

Reply:
Yes, makes perfect sense, I reworked the test accodingly.